met-api/src/met_api/resources/user.py

```diff
@@ -0,0 +1,57 @@
+# Copyright © 2021 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""API endpoints for managing an user resource."""
+
+from flask import g
+from flask_cors import cross_origin
+from flask_restx import Namespace, Resource
+
+from met_api.auth import auth
+from met_api.schemas.user import UserSchema
+from met_api.services.user_service import UserService
+from met_api.utils.util import allowedorigins, cors_preflight
+
+
+API = Namespace('user', description='Endpoints for User Management')
+"""Custom exception messages
+"""
+
+
+@cors_preflight('PUT')
+@API.route('/')
+class UserController(Resource):
+    """User controller class."""
+
+    @staticmethod
+    # @TRACER.trace()
+    @cross_origin(origins=allowedorigins())
+    @auth.require
+    def put():
+        """Update or create a user."""
+        try:
```

Comment:
hm.. good approach..
one challenge will be if we ever need a user update from a staff ;say staff updating users first name , this might not work..
a put on {user_id} will work in both cases..but we have to make sure the authorisation